src/client/event_handler.rs

```diff
@@ -9,7 +9,7 @@ use std::sync::RwLock;
 
 pub trait EventHandler {
     #[cfg(feature = "cache")]
-    fn on_cached(&self, _: Context, _: Vec<GuildId>) {}
```

Comment:
I should've specified that functions that users pass data unto us needed to be changed to use `IntoIterator` (since these're like the users' callbacks, just smashed into a trait), but oh well, revert these that you've done here.

Reply:
Oh right, so this will break users' code?

Reply:
Yes, since they'd have to rewrite their impl for that handler with the newly added trait bound; or else their code won't compile.

Reply:
Ok I see.